src/main/java/Barry.java

```diff
@@ -0,0 +1,322 @@
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class Barry {
+
+    // A class for elements in the taskList, with a name and an indicator of whether the task is marked
+    private abstract static class Task{
+        private final String taskName;
+        private boolean isDone;
+
+        // Constructor for the Task class, initialised with the name of the task
+        public Task(String name) {
+            this.taskName = name;
+            this.isDone = false;
+        }
+
+        // Allows the user to modify the status of the task
+        public void modifyStatus(boolean status) {
+            this.isDone = status;
+        }
+
+        // Returns the name of the task
+        protected final String getName() {
+            return this.taskName;
+        }
+
+        // Returns the status of the task
+        protected final boolean getStatus() {
+            return this.isDone;
+        }
```

Comment:
I think you could consider implementing your conditional statement in the getStatus() instead of returning boolean, as in the detailsAsString() method, you did the conditional statement in the additions of strings. I think it could perhaps be something like getStatusString() and you will get the string directly instead of having conditional statements in the additions of strings.